engine/dbscan/src/test/java/org/opennms/alec/engine/dbscan/DBScanEnginePerfTest.java

```diff
@@ -72,4 +88,68 @@ public void canRunDBScanOnLargeGraphs() {
             System.out.printf("%d ms for %d vertices.\n", delta, K);
         }
     }
+
+    @Test
+    public void canNotRunOOM() {
+        final DBScanEngine engine = new DBScanEngine();
+        engine.init(Collections.emptyList(), Collections.emptyList(), Collections.emptyList(),
+                Collections.emptyList());
+        engine.registerSituationHandler(mock(SituationHandler.class));
+
+        // With a 1GB heap these values would cause the cluster engine to run OOM before we swapped to using a soft
+        // reference backed implementation of Dijkstra's algorithm after ~50 seconds
+        //
+        // With the new impl this test passes after ~50 seconds
```

Comment:
Is it worth running this test on every commit if it adds 1 minutes to the build time?

Reply:
No, but as far as I know we don't have a way of picking which unit tests we want to run on-commit vs nightly currently. Do we want to look into adding something like that?

The alternative is we could just ignore it and only use it when we are trying to debug similar issues if they occur in the future.

Reply:
Given we don't have that support currently, maybe just @Ignore it for now.

Moving to JUnit 5 should give us the control we need by using the `@Tag` annotation, and skipping these with `-DexcludeTags={tag names}`.